QueryBuilder/Query.cs

```diff
@@ -336,12 +336,11 @@ public object FindVariable(string variable)
                 return Variables[variable];
             }
 
-            if (Parent != null)
+            if (Parent == null)
             {
-                return (Parent as Query).FindVariable(variable);
+                throw new Exception($"Variable '{variable}' not found");
             }
-
```

Comment:
When i look into issue 32 it looks like that issue also made the same change the commit id is 817fdb665bd7f704141e5d3b359a54a1bcc1067d